core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -341,6 +323,45 @@ public KeyStore getPrimaryKeyStore() throws Exception {
         }
     }
 
+    /**
+     * Load the requested tenant keystore from registry.
+     *
+     * @param keyStoreName  Name of the Tenant KeyStore.
+     * @return Key store object.
+     * @throws Exception    Exception if failed to retrive the given keystore from registry.
+     */
+    private KeyStore getTenantKeyStore(String keyStoreName) throws Exception {
+
+        if (isCachedKeyStoreValid(keyStoreName)) {
+            return tenantKeyStores.get(keyStoreName).getKeyStore();
+        }
+
+        String path = RegistryResources.SecurityManagement.KEY_STORES + "/" + keyStoreName;
+        if (registry.resourceExists(path)) {
+            org.wso2.carbon.registry.api.Resource resource = registry.get(path);
+            byte[] bytes = (byte[]) resource.getContent();
+            KeyStore keyStore = KeyStore.getInstance(resource
+                    .getProperty(RegistryResources.SecurityManagement.PROP_TYPE));
+            CryptoUtil cryptoUtil = CryptoUtil.getDefaultCryptoUtil();
+            String encryptedPassword = resource
+                    .getProperty(RegistryResources.SecurityManagement.PROP_PASSWORD);
+            String password = new String(cryptoUtil.base64DecodeAndDecrypt(encryptedPassword));
+            ByteArrayInputStream stream = new ByteArrayInputStream(bytes);
+            keyStore.load(stream, password.toCharArray());
+            KeyStoreBean keyStoreBean = new KeyStoreBean(keyStore, resource.getLastModified());
+            resource.discard();
+
+            if (tenantKeyStores.containsKey(keyStoreName)) {
+                tenantKeyStores.replace(keyStoreName, keyStoreBean);
+            } else {
+                tenantKeyStores.put(keyStoreName, keyStoreBean);
+            }
+            return keyStore;
+        } else {
+            throw new SecurityException("Key Store with a name : " + keyStoreName + " does not exist.");
```

Comment:
If possible suggest to use Java String formatters for multiple concatenations in single line :)
https://www.w3schools.com/java/ref_string_format.asp

Reply:
The existing code uses the '+' sign to concert strings. Also, agree with the point of moving to string formaters. But will keep the current formatting for now considering the consistency.